cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/CucumberExpression.java

```diff
@@ -19,16 +20,19 @@
     private final List<ParameterType<?>> parameterTypes = new ArrayList<>();
     private final String source;
     private final TreeRegexp treeRegexp;
+    private final ParameterTypeRegistry parameterTypeRegistry;
 
     public CucumberExpression(String expression, ParameterTypeRegistry parameterTypeRegistry) {
         this.source = expression;
+        this.parameterTypeRegistry = parameterTypeRegistry;
 
         expression = processEscapes(expression);
         expression = processOptional(expression);
         expression = processAlternation(expression);
         expression = processParameters(expression, parameterTypeRegistry);
         expression = "^" + expression + "$";
         treeRegexp = new TreeRegexp(expression);
+
```

Comment:
Unrelated to  this PR but we are using `StringBuffer` below. If we don't need threadsafety then `StringBuilder`is a better choice.

Reply:
And we can't do that until Java 9.